tools/common-code/cmonitor_launcher.py

```diff
@@ -0,0 +1,189 @@
+        proc_self_mount = "/proc/self/mounts"
+        ncgroup_v1 = 0
+        ncgroup_v2 = 0
+        with open(proc_self_mount) as file:
+            for line in file:
+                row = line.split()
+                fs_spec = row[0]
+                fs_file = row[1]
+                fs_vfstype = row[2]
+                if (
+                    fs_spec == "cgroup" or fs_spec == "cgroup2"
+                ) and fs_vfstype == "cgroup2":
+                    ncgroup_v2 += 1
+                else:
+                    ncgroup_v1 += 1
+
+        if ncgroup_v1 == 0 and ncgroup_v2 > 0:
+            cgroup_versopn = "v2"
+            return cgroup_version
+        else:
+            cgroup_version = "v1"
+            return cgroup_version
+
+    def get_process_name(self, pid):
+        cgroup_version = self.get_cgroup_version()
+        if cgroup_version == "v1":
+            proc_filename = "/proc" + "/" + pid + "/stat"
+        else:
+            proc_filename = "/proc" + "/" + pid + "/cgroup.procs"
+        with open(proc_filename) as file:
+            for line in file:
+                parts = line.split()
+                process_name = parts[1].strip("()")
+        return process_name
+
+    def get_pid_list(self, filename):
+        list = []
+        with open(filename) as file:
+            for line in file:
+                list.append(line.strip())
+        return list
+
+    def get_list_of_files(self, dir):
+        listOfFiles = os.listdir(dir)
+        allFiles = list()
+        for entry in listOfFiles:
+            fullpath = os.path.join(dir, entry)
+            if os.path.isdir(fullpath):
+                allFiles = allFiles + self.get_list_of_files(fullpath)
+            else:
+                allFiles.append(fullpath)
+
+        return allFiles
+
+    def process_task_files(self, dir):
+        time.sleep(5)
+        allFiles = self.get_list_of_files(dir)
+        for file in allFiles:
+            if file.endswith("tasks"):
+                list = self.get_pid_list(file)
+                if list:
+                    for pid in list:
+                        process_name = self.get_process_name(pid)
+                        match = self.check_filter(process_name)
+                        if match is True:
+                            print("Found match:", process_name)
+                            print("Launchig CMonitor", process_name, self.filter)
+                            self.launch_cmonitor(file)
+
+    def launch_cmonitor(self, filename):
+        for c in self.command:
+            cmd = c.strip()
+            base_path = os.path.dirname(filename)
+            path = "/".join(base_path.split("/")[5:])
+            cmd = cmd + " " + "--cgroup-name=" + path
+            print("Launch CMonitor : with command:", cmd)
+            # p = Popen(cmd)
+            # print("process id:",p.pid)  #FIXME : need to delete the current cmonitor already running fot this process..!!!
+            os.system(cmd)
```

Comment:
TBD : Need to find a way to trigger specific command for multiple events. currently command triggers with only a specific <ip> and <port> . Every command needs a separate <ip or a port> to avoid conflict.